src/backend/distributed/utils/citus_depended_object.c

```diff
@@ -480,8 +480,10 @@ AnyObjectViolatesOwnership(DropStmt *dropStmt)
 		Node *object = NULL;
 		foreach_ptr(object, dropStmt->objects)
 		{
+			Relation rel;
 			objectAddress = get_object_address(objectType, object,
-											   &relation, AccessShareLock, missingOk);
+											   &rel, AccessShareLock, missingOk);
+			relation = rel;
```

Comment:
I guess this is to only set the variable for closing if it's not this function that throws the error.

Is there a specific scenario where `get_object_address` does set the variable, but then can't close?

Would be great to capture the reason for separating the capture and assignment.

Reply:
Given it is set as the very last step, I don't see why we capture elsewhere on the stack to apply a reassign: https://github.com/postgres/postgres/blob/master/src/backend/catalog/objectaddress.c#L1208

Unless this is specific to `volatile` which even more so deserves a comment.

Reply:
Since the `relation` is qualified with `volatile` in this patch, the corresponding parameter of `get_object_address()` should be qualified with `volatile` as well. However, `get_object_address` is a function defined in the PostgreSQL server, we cannot change its signature.